Improve message on missing writer

When we throw an exception because of a missing writer or stream writer, the message is a bit cryptic:
```
"Transformation failed!"
Caused by: java.lang.IllegalStateException: No stream writer found for type: interface java.util.Map
	at io.helidon.media.common.MessageBodyWriterContext.marshallStream(MessageBodyWriterContext.java:355)
```

For a new user, this does not give any hint as what may be wrong. We may want to add an explanatory message, such as
"No stream writer found for type: interface java.util.Map, maybe `io.helidon.media:helidon-media-jsonb` or `io.helidon.media:helidon-media-jackson` should be added?" 

- maybe mentioning the `JsonbSupport` or `JacksonSupport` classes would be useful as well.

In general we should guide the user to find the place where to set this up (WebClient or WebServer builder) + how to set it up (module + class name).

--- a/media/common/src/main/java/io/helidon/media/common/MediaSupport.java
+++ b/media/common/src/main/java/io/helidon/media/common/MediaSupport.java
@@ -24,10 +24,16 @@ import java.util.Collections;
  * MediaSupport instances can be used with WebServer and WebClient to register readers and writer.
  * Each of these have method addMediaSupport(), which will add corresponding support.
  * </p><br>
- * Example usage:
+ * WebServer example usage:
  * <pre><code>
  * WebServer.builder()
- *          .addMediaSupport(MediaSupport)
+ *          .addMediaSupport(JsonbSupport.create())
+ *          .build();
+ * </code></pre>
+ * WebClient example usage:
+ * <pre><code>
+ * WebClient.builder()
+ *          .addMediaSupport(JacksonSupport.create())
  *          .build();
  * </code></pre>
  * If you need to register MediaSupport on the request or response, you will need to register them to

--- a/media/common/src/main/java/io/helidon/media/common/MediaSupport.java
+++ b/media/common/src/main/java/io/helidon/media/common/MediaSupport.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2020,2021 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -44,7 +44,7 @@ import java.util.Collections;
  * Routing.builder()
  *        .get("/foo", (res, req) -&gt; {
  *            MessageBodyReadableContent content = req.content();
- *            DefaultMediaSupport.create().readers().forEach(content::registerReader);
+ *            content.registerReader(JsonbSupport.create())
  *            content.as(String.class)
  *                   .thenAccept(System.out::print);
  *        })
@@ -54,11 +54,10 @@ import java.util.Collections;
  * Routing.builder()
  *        .get("/foo", (res, req) -&gt; {
  *           MessageBodyWriterContext writerContext = res.writerContext();
- *           DefaultMediaSupport.create().writers().forEach(writerContext::registerWriter);
+ *           writerContext.registerWriter(JsonbSupport.create())
  *           res.send("Example entity");
  *        })
  * </code></pre>
- * (DefaultMediaSupport is used only to illustrate the registration)
  */
 public interface MediaSupport {
 

--- a/media/common/src/main/java/io/helidon/media/common/MediaSupport.java
+++ b/media/common/src/main/java/io/helidon/media/common/MediaSupport.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020,2021 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2021 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java
+++ b/media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2020,2021 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java
+++ b/media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020,2021 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2021 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java
+++ b/media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java
@@ -305,7 +305,8 @@ public final class MessageBodyWriterContext extends MessageBodyContext implement
 
             MessageBodyWriter<T> writer = (MessageBodyWriter<T>) writers.select(type, this);
             if (writer == null) {
-                throw new IllegalStateException("No writer found for type: " + type);
+                throw new IllegalStateException("No writer found for type: " + type
+                        + ". This usually occurs when the appropriate MediaSupport has not been added.");
             }
             return applyFilters(writer.write(content, type, this));
         } catch (Throwable ex) {
@@ -352,7 +353,8 @@ public final class MessageBodyWriterContext extends MessageBodyContext implement
             }
             MessageBodyStreamWriter<T> writer = (MessageBodyStreamWriter<T>) swriters.select(type, this);
             if (writer == null) {
-                throw new IllegalStateException("No stream writer found for type: " + type);
+                throw new IllegalStateException("No stream writer found for type: " + type
+                        + ". This usually occurs when the appropriate MediaSupport has not been added.");
             }
             return applyFilters(writer.write(content, type, this));
         } catch (Throwable ex) {

--- a/tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/FormTest.java
+++ b/tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/FormTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2021 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/FormTest.java
+++ b/tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/FormTest.java
@@ -21,6 +21,7 @@ import io.helidon.common.http.MediaType;
 import org.junit.jupiter.api.Test;
 
 import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.startsWith;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
@@ -71,7 +72,7 @@ public class FormTest extends TestParent {
                 .submit(TEST_FORM).await());
 
         assertThat(ex.getCause().getMessage(),
-                   is("No writer found for type: class io.helidon.common.http.FormParamsImpl"));
+                   startsWith("No writer found for type: class io.helidon.common.http.FormParamsImpl"));
     }
 
     @Test

